airflow/models.py

```diff
@@ -4062,7 +4072,11 @@ def verify_integrity(self, session=None):
             try:
                 dag.get_task(ti.task_id)
             except AirflowException:
-                if self.state is not State.RUNNING:
+                # TODO (bolke): a backfill can consider a subset of the original
+                # tasks of a whole dag but with the same name. If that is the case
+                # we should not mark tasks removed as we might not consider
+                # the whole dag.
+                if self.state is not State.RUNNING and not self.is_backfill:
```

Comment:
It's a common use case to mutate the DAG locally in the hopes of generating a custom backfill. In those cases the user has the assumption that Airflow only knows about the tasks defined. The user would expect state changes only on tasks that are in scope.